inst/include/dust/random/normal_ziggurat.hpp

```diff
@@ -0,0 +1,75 @@
+#ifndef DUST_RANDOM_NORMAL_ZIGGURAT_HPP
+#define DUST_RANDOM_NORMAL_ZIGGURAT_HPP
+
+#include <cmath>
+
+#include "dust/random/generator.hpp"
+#include "dust/random/normal_ziggurat_tables.hpp"
+
+namespace dust {
+namespace random {
+
+__nv_exec_check_disable__
+namespace {
+template <typename real_type, typename rng_state_type>
+__host__ __device__
+real_type normal_ziggurat_tail(rng_state_type& rng_state, real_type x1,
+                               bool negative) {
+  real_type ret;
+  do {
+    const auto u1 = random_real<real_type>(rng_state);
+    const auto u2 = random_real<real_type>(rng_state);
+    const auto x = std::log(u1) / x1;
+    const auto y = std::log(u2);
+    if (- 2 * y > x * x) {
+      ret = negative ? x - x1 : x1 - x;
+      break;
+    }
+  } while (true);
+  return ret;
+}
+}
+
+// TODO: this will not work efficiently for float types because we
+// don't have float tables for 'x' and 'y'; getting them is not easy
+// without requiring c++14 either. The lower loop using 'x' could
+// rewritten easily as a function though taking 'u1' and so allowing
+// full template specialisation. However by most accounts this
+// performs poorly onn a GPU to latency so it might be ok.
+__nv_exec_check_disable__
+template <typename real_type, typename rng_state_type>
+__host__ __device__
+real_type random_normal_ziggurat(rng_state_type& rng_state) {
+  using ziggurat::x;
+  using ziggurat::y;
+  constexpr size_t n = 256;
```

Comment:
Comment to explain this choice and whether it is tuneable?

Reply:
My first shot at this was fully tunable using `std::array<real_type, int>` for all sorts  of useful `n`! But it was a pain to work with and we can't have `std::array` on the GPU...

Reply:
I've added a note indicating where this comes from and why now